### DISCUSSION TITLE
Is there a way to differentiate an island from a hole for a polygon?

### DISCUSSION THREAD

**Discussion by yingdanh** [alexbol99/flatten-js#117] [Q&A]
We have several polygons to perform a unify function, which got islands and holes. Is there a way to figure out islands from holes?

**Comment by alexbol99** [comment]
Hello, @yingdanh 

In ```flatten-js``` library, polygon contains loops (faces), and the orientation of the loop defines if polygon is island or hole.
The main idea that "islands" and "holes" should have opposite orientation, and outermost loop always considered as "island".
User defines which orientation he wants for islands - clockwise or counterclockwise.
In Boolean operations each polygon should have same orientation for islands, this is important.

For example, if you want to add polygon as a hole into an island, you have to define both of of them as islands and then use ```subtract``` operation. The resulted polygon will have two loops: one island and one hole.

I hope this explanation will be helpful
